Find a functionally equivalent code:com.android.internal.widget.LockPatternUtils.getNextAlarm:COMMENT
Method Modifier: public      internal    
Comment:/**
 * @return A formatted string of the next alarm (for showing on the lock screen),
 * or null if there is no next alarm.
 */

@@ -1,7 +1,4 @@
 {
-    String nextAlarm = Settings.System.getStringForUser(mContentResolver, Settings.System.NEXT_ALARM_FORMATTED, UserHandle.USER_CURRENT);
-    if (nextAlarm == null || TextUtils.isEmpty(nextAlarm)) {
-        return null;
-    }
-    return nextAlarm;
+    AlarmManager alarmManager = (AlarmManager) mContext.getSystemService(Context.ALARM_SERVICE);
+    return alarmManager.getNextAlarmClock(UserHandle.USER_CURRENT);
 }

